internal/argocd/revision.go

```diff
@@ -1,35 +1,81 @@
 package argocd
 
 import (
+	"context"
+	"errors"
 	"path"
 
 	kargoapi "github.com/akuity/kargo/api/v1alpha1"
 	argocd "github.com/akuity/kargo/internal/controller/argocd/api/v1alpha1"
 	"github.com/akuity/kargo/internal/git"
+	"github.com/akuity/kargo/internal/logging"
 )
 
 // GetDesiredRevision returns the desired revision for the given
 // v1alpha1.Application by traversing the given v1alpha1.FreightReference for
 // a matching source. If no match is found, an empty string is returned.
 func GetDesiredRevision(app *argocd.Application, freight kargoapi.FreightReference) string {
-	switch {
-	case app == nil || app.Spec.Source == nil:
-		// Without an Application, we can't determine the desired revision.
+	ctx := context.Background()
```

Comment:
As this is a utility function, I would not introduce logging at this level but rather ensure this is handled by the caller.

Reply:
When I was debugging, it was super difficult without this logging to isolate why my app was not being considered for update.

If we follow your suggestion and remove logging here, what does it add to the overall solution in exchange for reducing the optional visibility into the decision chain?

Reply:
I don't have a strong opinion about whether logging _belongs_ here or not, but whatever is decided, please use a context that was passed to the function instead of grabbing a new background context. The benefit of that is that if a caller has added details to the logger and stashed that updated logger in the context, this function will inherit that logger with those details.

Reply:
@hiddeco , if you're okay with having debug logs, I'd make the requested updates to this. This would really help users debug their own deployments without having to modify and run the code like I have.

Reply:
I won't be a blocker here.